lib/network/utils/RoundRobin.js

```diff
@@ -0,0 +1,128 @@
+const DEFAULT_STICKY_COUNT = 100;
+
+/**
+ * Shuffle an array in-place
+ *
+ * @param {Array} array - The array to shuffle
+ * @return {undefined}
+ */
+function shuffle(array) {
+    for (let i = array.length - 1; i > 0; i--) {
+        const randIndex = Math.floor(Math.random() * (i + 1));
+        /* eslint-disable no-param-reassign */
+        const randIndexVal = array[randIndex];
+        array[randIndex] = array[i];
+        array[i] = randIndexVal;
+        /* eslint-enable no-param-reassign */
+    }
+}
+
+class RoundRobin {
+    /**
+     * @constructor
+     * @param {object[]} hostsList - list of hosts to query in
+     *   round-robin fashion
+     * @param {string} hostsList[].host - host name or IP address
+     * @param {number} hostsList[].port - port number to contact
```

Comment:
Why not, technically the round-robin logic does not need items to have any specific fields (items like `{ banana: 42 }` would work as well), the sanity checks done are also part of the helper logic to make sure the user provides valid input, so we can be quite flexible on this.

One way of dealing with missing ports could be to provide a default port value at construction, which would get appended in case the item has no port specified. Or we can let the application deal with missing port if it does not require it.